src/java.base/share/classes/sun/nio/ch/SelChImpl.java

```diff
@@ -83,13 +82,17 @@ public interface SelChImpl extends Channel {
      * @param nanos the timeout to wait; {@code <= 0} to wait indefinitely
      */
     default void park(int event, long nanos) throws IOException {
-        long millis;
-        if (nanos <= 0) {
-            millis = -1;
+        if (Thread.currentThread().isVirtual()) {
+            Poller.poll(getFDVal(), event, nanos, this::isOpen);
         } else {
-            millis = NANOSECONDS.toMillis(nanos);
+            long millis;
+            if (nanos == 0) {
```

Comment:
Was this change intentional? (`<=` replaced by `==`) 
If it was, please update the javadoc - `NANOSECONDS.toMillis(-1)` = 0 implies no waiting in Net.poll

Reply:
> If it was, please update the javadoc - `NANOSECONDS.toMillis(-1)` = 0 implies no waiting in Net.poll

It's benign for the current usages but you are right, it was not intentional.